nifpga/session.py

```diff
@@ -390,8 +413,250 @@ def read(self):
+        point is none zero that indicates the data read has overflowed. """
+        mask = 2**(self._word_length)
+        if data & mask > 0:
+            return True
+        return False
+
+    def _remove_overflow_bit(self, data):
+        """ This helper method masks out all bits not inside the word length,
+        ultimately returning a value of data without the overflow bit. """
+        return data & (2**(self._word_length) - 1)
+
+    def _integer_twos_comp(self, data):
+        if (data & (1 << (self._word_length - 1))) != 0:
+            data = data ^ (2 ** (self._word_length) - 1)
+            data += 1
+            data *= -1
+        return data
+
+    def write(self, user_input):
+        """ Writes the user's the users input into the register as a fixed
+        point number. Any inputs outside the bounds of this fixed point
+        register will be coerced and the user will be warned. The user input's
+        supported include any python Number.
+
+            Args:
+                Number: user numerical input to be converted to fixed point.
+                (bool, Number): Tuple with the members : Boolean for overflow,
+                                user numerical input to be converted to fixed
+                                point.
+        """
+
+        fxp_representation = self._convert_user_input_to_fxp_representation(user_input)
+        arrayData = self._extract_array_of_u32_from_fxp_value(fxp_representation)
+        buf = self._ctype_type(*arrayData)
+        self._write_func(self._session, self._resource, buf, self._transfer_len)
+
+    def _convert_user_input_to_fxp_representation(self, user_input):
+        (overflow, data) = self._validate_and_parse_user_input(user_input)
+
+        fxp_representation = 0
+        if data < self._minimum:
+            fxp_representation = self._convert_value_to_fxp(self._minimum)
+            self.warn_coerced_data()
+        elif data > self._maximum:
+            fxp_representation = self._convert_value_to_fxp(self._maximum)
+            self.warn_coerced_data()
+        else:
+            fxp_representation = self._convert_value_to_fxp(data)
+
+        if self._signed and data < 0:
+            fxp_representation = self._integer_twos_comp(fxp_representation)
+
+        if self._overflow_enabled:
+            if overflow:
+                fxp_representation += 2**(self._word_length)
+
+        return fxp_representation
+
+    def _validate_and_parse_user_input(self, user_input):
+        overflow = None
+        data = None
+        if self._overflow_enabled:
+            (overflow, data) = user_input
+            assert isinstance(overflow, bool)
+        else:
+            data = user_input
+        assert isinstance(data, Number)
+        return (overflow, data)
+
+    def _convert_value_to_fxp(self, data):
+        calculated_fxp = Decimal(data) / Decimal(self._delta)
+        fxp_representation = int(calculated_fxp)
```

Comment:
You might need to do a "long" here rather than int.  int is only 32 bits in 32bit Python.

Reply:
It is actually smart enough to do the conversion to long if it needs to. On python 2.7 this is what I get. Plus the unit tests would of caught this if something changes in a later version.
>>> int(1357135617236725416357)
1357135617236725416357L
>>> int(150)
150
>>>